mob_test.go

```diff
@@ -76,70 +76,50 @@ func assertDetermineBranches(t *testing.T, branch string, qualifier string, bran
 
-func TestEnvironmentVariablesEmptyString(t *testing.T) {
-	os.Setenv("MOB_REMOTE_NAME", "")
-	defer os.Unsetenv("MOB_REMOTE_NAME")
-
-	configuration = parseEnvironmentVariables(getDefaultConfiguration())
+func TestMobRemoteNameEnvironmentVariableEmptyString(t *testing.T) {
+	configuration = setEnv("MOB_REMOTE_NAME", "")
 
 	equals(t, "origin", configuration.RemoteName)
 }
 
-func TestMobDoneSquashEnvironmentVariableDefault(t *testing.T) {
-	configuration = parseEnvironmentVariables(getDefaultConfiguration())
-
-	equals(t, true, configuration.MobDoneSquash)
-}
-
-func TestMobDoneSquashEnvironmentVariableEmpty(t *testing.T) {
-	// ASSUME default setting for mobDoneSquash is true
-	os.Setenv("MOB_DONE_SQUASH", "")
-	defer os.Unsetenv("MOB_DONE_SQUASH")
-
-	configuration = parseEnvironmentVariables(getDefaultConfiguration())
-
-	equals(t, true, configuration.MobDoneSquash)
+func TestBooleanEnvironmentVariables(t *testing.T) {
+	expectEnvironmentVariableSetsBooleanConfiguration(t, "MOB_DONE_SQUASH", true, func(configuration Configuration) interface{} { return configuration.MobDoneSquash })
+	expectEnvironmentVariableSetsBooleanConfiguration(t, "MOB_DEBUG", false, func(configuration Configuration) interface{} { return configuration.Debug })
 }
 
-func TestMobDoneSquashEnvironmentVariableTrue(t *testing.T) {
-	os.Setenv("MOB_DONE_SQUASH", "true")
-	defer os.Unsetenv("MOB_DONE_SQUASH")
-
-	configuration = parseEnvironmentVariables(getDefaultConfiguration())
-
-	equals(t, true, configuration.MobDoneSquash)
+func expectEnvironmentVariableSetsBooleanConfiguration(t *testing.T, envVar string, defaultValue bool, actual func(Configuration) interface{}) {
+	t.Run(envVar, func(t *testing.T) {
+		expectEnvironmentVariableSetsConfiguration(t, envVar, "", defaultValue, actual)
+		expectEnvironmentVariableSetsConfiguration(t, envVar, "true", true, actual)
+		expectEnvironmentVariableSetsConfiguration(t, envVar, "false", false, actual)
+		expectEnvironmentVariableSetsConfiguration(t, envVar, "garbage", defaultValue, actual)
+	})
 }
 
-func TestMobDoneSquashEnvironmentVariableFalse(t *testing.T) {
-	os.Setenv("MOB_DONE_SQUASH", "false")
-	defer os.Unsetenv("MOB_DONE_SQUASH")
-
-	configuration = parseEnvironmentVariables(getDefaultConfiguration())
-
-	equals(t, false, configuration.MobDoneSquash)
+func expectEnvironmentVariableSetsConfiguration(
+	t *testing.T,
+	variable string,
+	value string,
+	expected interface{},
+	actual func(Configuration) interface{},
+) {
+	t.Run(variable+"=\""+value+"\"(="+fmt.Sprintf("%t", expected)+")", func(t *testing.T) {
+		configuration = setEnv(variable, value)
+		equals(t, expected, actual(configuration))
+	})
 }
 
-func TestMobDoneSquashEnvironmentVariableGarbage(t *testing.T) {
-	// ASSUME default setting for mobDoneSquash is true
-	os.Setenv("MOB_DONE_SQUASH", "garbage")
-	defer os.Unsetenv("MOB_DONE_SQUASH")
+func setEnv(variable string, value string) Configuration {
+	os.Setenv(variable, value)
+	defer os.Unsetenv(variable)
```

Comment:
I think I would rename the function to something like `setAndParseEnvironmentVariableInConfiguration` to make it really clear what happens here. What do you think?